pythonwhat/checks/has_funcs.py

```diff
@@ -349,15 +349,19 @@ def has_expr(
         if isinstance(eval_sol, str):
             fmt_kwargs["sol_eval"] = '\'{}\''.format(fmt_kwargs["sol_eval"])
 
+    # reformat student evaluation string if it is too long or contains newlines
+    if incorrect_msg == DEFAULT_INCORRECT_MSG:
+        fmt_kwargs["stu_eval"] = utils.shorten_string(fmt_kwargs["stu_eval"])
+
     # check if student or solution evaluations are too long or contain newlines
     if incorrect_msg == DEFAULT_INCORRECT_MSG and (
```

Comment:
This line could be removed and everything moved under the if above (if that if isn't removed), also the comment above this line is no longer completely correct.